Formula/bazelisk.rb

```diff
@@ -25,6 +25,7 @@ def install
   end
 
   test do
+    ENV["USE_BAZEL_VERSION"] = version.to_s
```

Comment:
Bazelisk is not versioned together with Bazel, so it's not safe to assume that a Bazelisk version (`version.to_s`) is also a valid Bazel version (`$USE_BAZEL_VERSION`).

I would just set this to "1.1.0" for now to avoid the broken 1.2.0 release and add a comment why that's ok. The first assert only tests that Bazelisk can print its own version number correctly, so the Bazel version that it downloads doesn't really matter.

Reply:
That `version` is actually the bazelisk version. So technically, that is "1.1.0" though.

I am trying to avoid the hardcoded value that there would be less change in terms of maintenance efforts.

Reply:
Yes, maybe I'm misunderstanding something here, I'm not so used to Homebrew recipes yet. 🤔

`ENV["USE_BAZEL_VERSION"]` has to specify a Bazel version (that environment variable is used by Bazelisk here: https://github.com/bazelbuild/bazelisk/blob/master/bazelisk.go#L80).

However, because we're in the `bazelisk.rb` here, the `version.to_s` does not refer to a Bazel version, but instead refers to the Bazelisk version defined above in line 5 via the `:tag`, right?

It is just by chance that we currently have Bazel 1.1.0 and Bazelisk 1.1.0 at the same time - usually these tools do not have comparable version numbers.

Reply:
yeah, the version is referring to the version in that formula. so in that sense it would be 1.1.0 (as that is the version for bazelisk)

Reply:
Let me know if that makes sense now.

Reply:
Considering 1.2.1 is now released does it even matter now? Do we still need this?

Reply:
I guess it is fine. All it does is just to download the latest bazel artifact.

Reply:
Sorry to come back to this, but I still think this is not correct: Unless I'm completely missing something, you cannot use `version.to_s` here. I don't know how else to say this. `version.to_s` in this formula (bazelisk.rb) refers to the **Bazelisk** version. But the content of the `USE_BAZEL_VERSION` environment variable refers to the **Bazel** version that Bazelisk should download. Bazelisk vs. Bazel. The two tools do not share version numbers. So, you cannot use `ENV["USE_BAZEL_VERSION"] = version.to_s`, because the left side and the right side do not refer to the same version numbers.

The fact that this works at the moment is pure luck, because the latest version for both tools is 1.1.0. When I release Bazelisk 2.0.7a "Moonshot Edition" (just an example) tomorrow and you update this formula to match, this will break, because there is no "Bazel 2.0.7a" and will never be, but by setting `USE_BAZEL_VERSION` to `version.to_s` (which will be "2.0.7a" then), Bazelisk will try to download "Bazel 2.0.7a" which doesn't exist.

Sorry to be so pedantic about this, but I just worry that this test will break at the next version bump and then people spend time debugging this and wonder what's going on and why Bazelisk doesn't work.

Reply:
```suggestion
    ENV["USE_BAZEL_VERSION"] = Formula["bazel"].version
```

Reply:
that is actually good very suggestion. Let me Adopt it!